odoo/addons/base/models/res_partner.py

```diff
@@ -451,21 +451,22 @@ def _fields_sync(self, values):
         self._children_sync(values)
 
     def _children_sync(self, values):
-        if not self.child_ids:
+        child_ids = self.with_context(active_test=False).search([('parent_id', '=', self.id)])
```

Comment:
here you should not need the active_test because you match the override of _search:

```py
    @api.model            
    def _search(self, args, offset=0, limit=None, order=None, count=False, access_rights_uid=None):
        """ Override search() to always show inactive children when searching via ``child_of`` operator. The ORM will
        always call search() with a simple domain of the form [('parent_id', 'in', [ids])]. """
        # a special ``domain`` is set on the ``child_ids`` o2m to bypass this logic, as it uses similar domain expressions
        if len(args) == 1 and len(args[0]) == 3 and args[0][:2] == ('parent_id','in') \
                and args[0][2] != [False]:
            self = self.with_context(active_test=False)
```

but maybe it's a little too magic, so looks ok to keep it to me